### PR TITLE
Feature/search menu discover v1

### PR DESCRIPTION
**Pull request by ntissieres**
Feature available into the search menu, and that display the hierarchy of tags in which the user can dive and explore new hobby, while the list of events updates accordingly

### CODE REVIEW

**Review by unglazedstamp — CHANGES_REQUESTED** [review]
Good code ! I noticed that it's not possible to expand the bottom sheet, is it on purpose ? I think that it could be useful to expand it to have more space since the boxes of tags are big.
Why is the search interest text input still editable ? I believe that it should be disabled in the discovery mode if it doesn't have a function the discovery process.

File: app/src/main/java/com/github/swent/echo/compose/components/searchmenu/SearchMenuDiscover.kt [hunk 1]
```diff
@@ -1,5 +1,192 @@
...
+                            }
+                        )
+                )
+            }
+        }
+        Spacer(modifier = Modifier.height(5.dp))
+        // Display the tags
+        LazyVerticalGrid(
+            columns = GridCells.Fixed(2),
+            contentPadding = PaddingValues(start = 16.dp, top = 16.dp, end = 16.dp, bottom = 16.dp),
```
[the remark this answers is not in the document]
Comment (ntissieres) [reply app/src/main/java/com/github/swent/echo/compose/components/searchmenu/SearchMenuDiscover.kt:148]: Yes i could do that !

**Comment by ntissieres** [comment]
The bottom sheet is not expandable on purpose, the idea is that we want to see the events on the map or on the list while diving into the hierarchy. Moreover, the expanse would be usefull for discover but not for filters, and since they both lay on the same bottom sheet, it's a bit a mess to handle different mode. It could be good to also let the main sheet interactible while the bottom sheet is up, but it is not implemented yet.
What I can do for now is to set the size of the boxes smaller. Since we don't display subtags for this version, it's not necessary to have such big boxes ;)

File: app/src/main/java/com/github/swent/echo/compose/components/searchmenu/SearchMenuDiscover.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ntissieres) [reply app/src/main/java/com/github/swent/echo/compose/components/searchmenu/SearchMenuDiscover.kt:148]: or better just write PaddingValues(16.dp), I thought it was necessary to specify for each direction but apparently not

**Review by unglazedstamp — APPROVED** [review]
Thank you for the explanation and the corrections !

**Pull request merged**